### DISCUSSION TITLE
Show some content only in lastpage

### DISCUSSION THREAD

**Discussion by wesleigomes** [QuestPDF/QuestPDF#47] [Q&A] [answered]
Hi, I would like to show some content only in lastpage. Some tip?

**Comment by MarcinZiabek** [comment]
Where exactly would you like to show this content? In header/footer? On a separate and unrelated page? Please give an example

**Reply by wesleigomes** [reply]
In footer of the last page. Illustration of the case.

![Illustration](https://user-images.githubusercontent.com/16633920/138900143-06886629-7ac9-4708-8e93-313df33af766.png)

**Comment by MarcinZiabek** [comment, marked as answer]
I assume, that in the page content, a top-level container is a Stack. Then, as the last element of this Stack, please put an AlignBottom with your desired content. Let me know if it works, didn't try it :D

**Reply by wesleigomes** [reply]
Last page, ok. But the Content is not at the footer. The result is showed below.

`void ComposeContent(IContainer container)
        {
            container.Stack(stack =>
            {
                stack.Item().Component<TituloConteudo>(new(Model.TituloConteudo));

                GerarTituloConteudo(stack);

                stack.Item().Component<BasicData>(new(data.Basic));                
                stack.Item().Component<TableOfContent>(new(data.Content));
                stack.Item().AlignBottom().Component<ContentFooter>(new(data.Footer));
            });
        }`

![Illustration](https://user-images.githubusercontent.com/16633920/138906887-85dfc1ec-6228-46b7-ab0b-731b17654c47.png)

**Reply by MarcinZiabek** [reply]
Please try this:

```csharp
container.Page(page =>
{
    page.Margin(50);
    page.Background(Colors.White);
    
    page.Content().Stack(stack =>
    {
        stack.Spacing(20);
        
        foreach (var _ in Enumerable.Range(1, 23))
        {
            stack
                .Item()
                .Background(Colors.Grey.Lighten2)
                .Height(50);
        }

        stack
            .Item()
            .ExtendVertical() // <- important
            .AlignBottom() // <- important
            .Background(Colors.Green.Lighten2)
            .ExtendHorizontal()
            .Height(100);
    });
    
    page.Footer()
        .Background(Colors.Red.Lighten2)
        .Height(100);
});
```

Result:
![image](https://user-images.githubusercontent.com/9263853/138909479-a29f4e5d-6257-4f1e-a488-282b75be7dd2.png)

Future versions of the library may provide more tools regarding such layouts.

**Comment by wesleigomes** [comment]
Great!!! I didn't know ExtendVertical.

I want to congratulate you for the initiative. QuestPDF is fantastic and I think it has the potential to become a new world reference for .NET reporting.

Thanks!

**Reply by MarcinZiabek** [reply]
Thank you! Such words really give a person the reason why it is worth spending so much time on a side project. I really hope that at some point, the library will grow and find more users. It obviously has its own limitations, mainly derived from SkiaSharp, but still more and more useful with each release.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
